app/controllers/panel_controller.rb

```diff
@@ -46,4 +46,12 @@ def generate_db_token
       replica: 2,
     }
   end
+
+  def redirect
+    @panel_page = params.require(:panel_page)
+    panel_with_panel_page = current_user&.panels_with_access&.find { |panel| User.panel_list[panel][:pages].include?(@panel_page) }
```

Comment:
Why are you using nil-safe access here? You have `before_action :authenticate_user` in this controller already, so I don't see how `current_user` could ever be nil.

Reply:
Oh yes, I've removed it now. Thanks for pointing it out.